packages/next/src/lib/metadata/resolvers/resolve-url.ts

```diff
@@ -4,6 +4,25 @@ function isStringOrURL(icon: any): icon is string | URL {
   return typeof icon === 'string' || icon instanceof URL
 }
 
+function createLocalMetadataBase() {
+  return new URL(`http://localhost:${process.env.PORT || 3000}`)
+}
+
+// For deployment url for metadata routes, prefer to use the deployment url if possible
+// as these routes are unique to the deployments url.
+export function getFallbackMetadataBaseIfPresent(
+  metadataBase: URL | null
+): URL | null {
+  const defaultMetadataBase = createLocalMetadataBase()
+  return process.env.NODE_ENV === 'production' &&
+    process.env.VERCEL_URL &&
+    process.env.VERCEL_ENV === 'preview'
```

Comment:
What about production deployments? If I have `metadataBase` missing, it will be `VERCEL_URL` on preview deployments BUT `http://localhost:3000` on production deployments? 😨

Reply:
For prod case we should pick metadataBase but fallback to VERCEL_URL 🤔 going to fix it in another PR